Add environment variables to support bundle (#2073)
grafana/alloy: docs/sources/reference/cli/environment-variables.md

* `GODEBUG`
* `HTTP_PROXY`
* `HTTPS_PROXY`
* `NO_PROXY`
* `PPROF_MUTEX_PROFILING_PERCENT`
* `PPROF_BLOCK_PROFILING_RATE`
* `GOMEMLIMIT`
* `AUTOMEMLIMIT`
* `GOGC`
* `GOMAXPROCS`
* `GOTRACEBACK`

Refer to the [Go runtime][runtime] documentation for more information about Go runtime environment variables.

> # Windows Service Configuration

Reviewer: ```suggestion
## Windows Service Configuration
```

We can only have one H! heading per topic.

Also... do we really need this Windows-specific info here? What about Linux and MacOS? If we add specific instructions for Windows we should consider the same for the other supported OSes.

Author: Removed the windows specific information for now.

Reviewer: I added this to a liked issue so I don't forget to eventually add the Env Var details.

> To configure any of the available environment variables when running {{< param "PRODUCT_NAME" >}} as a Windows Service, open `regedit`, the Windows Registry Editor, and navigate to the registry key `HKEY_LOCAL_MACHINE\SOFTWARE\GrafanaLabs\Alloy`. Under this key will be an `Environment` multi-string value where any desired environment variables can be provided on separate lines.

Reviewer: It might be better that we refer to the configure topic for these steps instead of duplicating them here.

Something like this:

```
Refer to [Change environment variable values][change] for more information about changing the environment variables in Windows.

[change]: https://grafana.com/docs/alloy/latest/reference/cli/environment-variables/#change_environment_variable_values
```

If we add Linux/MacOS config instructions (like the Windows instructions) then we'd probably need to rework this a bit here

> For example, if you wanted to configure `NO_PROXY` and `GOGC` you would enter the following as the `Environment` value. Refer to our [Windows] configuration documentation for more details.

Reviewer: I am not sure this example belongs here. I'd suggest it's in the OS-specific config topic - ie in the Windows topic.

#### Automatically set GOMEMLIMIT

> The `GOMEMLIMIT` environment variable is either automatically set to 90% of an available `cgroup` value using the [automemlimit] module, or you can explicitly set the  `GOMEMLIMIT` environment variable before you run  {{< param "PRODUCT_NAME" >}}. The 90% ratio can also be changed by setting the `AUTOMEMLIMIT` environment variable to a float value between `0` and `1.0`.

Reviewer: ```suggestion
The `GOMEMLIMIT` environment variable is either automatically set to 90% of an available `cgroup` value using the [automemlimit] module, or you can explicitly set the  `GOMEMLIMIT` environment variable before you run  {{< param "PRODUCT_NAME" >}}.
You can also change the 90% ratio by setting the `AUTOMEMLIMIT` environment variable to a float value between `0` and `1.0`.
```

No changes will occur if the limit cannot be determined and you did not explicitly define a  `GOMEMLIMIT` value.

## GOGC

The `GOGC` environment variable controls the mechanism that triggers Go's garbage collection.
It represents the garbage collection target percentage. A collection is triggered when the ratio 
of freshly allocated data to live data remaining after the previous collection reaches this percentage.

> The default value if this variable is not provided is 100. Setting GOGC=off can also disable garbage collection entirely.

Reviewer: ```suggestion
If you don't provide this variable, GOGC defaults to `100`.
You can set `GOGC=off` to disable garbage collection.
```

Configuring this value in conjunction with `GOMEMLIMIT` can help in situations where {{< param "PRODUCT_NAME" >}} 
is consuming too much memory. Go provides a [very in-depth guide][gc_guide] to understanding `GOGC` and `GOMEMLIMIT`.

## GOMAXPROCS

The `GOMAXPROCS` environment variable defines the limit of OS threads that can simultaneously execute user-level Go code.
This limit does not affect the number of threads that can be blocked in system calls on behalf of Go code and those
threads are not counted against `GOMAXPROCS`.

## GOTRACEBACK

> The `GOTRACEBACK` environment variable defines the behavior of the Go panic output. The standard panic output behavior is usually
> sufficient to debug and resolve an issue, but this setting can be used in cases of a reproducable panic to collect
> additional information from the runtime if required. The following values are supported.

Reviewer: ```suggestion
The `GOTRACEBACK` environment variable defines the behavior of the Go panic output.
The standard panic output behavior is usually sufficient to debug and resolve an issue.
If required, you can use this setting to collect additional information from the runtime.
The following values are supported.
```

Value            | Description                                                                     | Traces include runtime internal functions
-----------------|---------------------------------------------------------------------------------|------------------------------------------
 `none` or `0`   | Omit goroutine stack traces entirely from the panic output.                     | -
 `single`        | Print the stack trace for the current goroutine.                                | No

>  `all` or `1`    | Print the stack traces for all user created goroutines.                         | No
>  `system` or `2` | Print the stack traces for all user created and runtime created goroutines.     | Yes

Reviewer: ```suggestion
 `all` or `1`    | Print the stack traces for all user-created goroutines.                         | No
 `system` or `2` | Print the stack traces for all user-created and runtime-created goroutines.     | Yes
```

>  `crash`         | Similar to `system`, but also triggers OS specific additional behavior. For example, on Unix systems this raises a SIGABRT to trigger a code dump.   | Yes

Reviewer: ```suggestion
 `crash`         | Similar to `system`, but also triggers OS-specific additional behavior. For example, on Unix systems, this raises a SIGABRT to trigger a code dump.   | Yes
```
